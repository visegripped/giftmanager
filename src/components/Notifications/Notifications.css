.message {
  padding: 20px;
  text-align: left;
  border: 1px solid;
  position: relative;
  border-radius: 4px;
  margin-bottom: 8px;
}

.message.info {
  background: lightblue;
  border-color: darkblue;
}

.message.warn {
  background: lightgoldenrodyellow;
  border-color: darkorange
}

.message.error {
  background: lightpink;
  border-color: darkred;
}

.message--pre {
  font-size: 14px;
  color: #333;
  overflow-x: auto;
  white-space: pre-wrap;
  word-wrap: break-word;
}

.message--button {
  position: absolute;
  top: 4px;
  right: 4px;
}