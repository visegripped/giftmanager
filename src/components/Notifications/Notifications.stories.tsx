import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import Notifications from './index';
// import { IMessage } from '../../context/NotificationContext';

// const sampleMessages: IMessage[] = ;

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'component/Notifications',
  component: Notifications,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
} as ComponentMeta<typeof Notifications>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Notifications> = () => <Notifications />;

export const BasicUsage = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
BasicUsage.args = {
  messages: [
    // { id: '1234-abcd-error', type: 'error', report: 'this is an error notification.' },
  ],
};
