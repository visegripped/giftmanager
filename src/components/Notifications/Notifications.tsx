import React from 'react';
import './Notifications.css';
import { IMessage } from '../../context/NotificationContext';
import 'Notifications.css';

interface NotificationsProps {
  cssClasses?: string;
  messages: [];
}

export const Message = (props: IMessage) => {
  const { type, report, id } = props;
  const handleClick = (clickEvent: React.SyntheticEvent<HTMLButtonElement>) => {
    const messageId = clickEvent.currentTarget?.dataset?.messageid;
    // removeMessage(messageId);
    console.log(`Click triggered for removing ${messageId}`);
  };
  return (
    <div className={`message ${type}`}>
      <pre className="message--pre">{report}</pre>
      <button className={`button message--button message--button--${type}`} data-messageid={id} onClick={handleClick}>
        X
      </button>
    </div>
  );
};

const Notifications = (props: NotificationsProps) => {
  const { cssClasses, messages } = props;
  const className = `Notifications ${cssClasses}`;
  return (
    <section className={className}>
      {messages.map((message: IMessage) => {
        return <Message {...message} />;
      })}
    </section>
  );
};

export default Notifications;
