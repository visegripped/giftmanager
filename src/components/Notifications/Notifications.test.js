import { render, screen } from '@testing-library/react';
import Notifications from './index';
import { Message } from './Notifications';

const messages = [
  { id: '1234-abcd-error', type: 'error', report: 'this is an error notification.' },
  { id: '1234-abcd-info', type: 'info', report: 'this is an info notification.' },
  { id: '1234-abcd-warn', type: 'warn', report: 'this is an warn notification.' },
  { id: '1234-abcd-success', type: 'success', report: 'this is an success notification.' },
  { id: '1234-abcd-standard', type: 'standard', report: 'this is an standard notification.' },
];

describe('Notifications', () => {
  test('snapshot matches', () => {
    render(<Notifications />);
    const renderedNotifications = screen.getByTestId('Notifications');
    expect(renderedNotifications).toMatchSnapshot();
  });
});

describe('Message', () => {
  test('snapshot for error state matches', () => {
    render(<Message {...messages[0]} />);
    const renderedNotifications = screen.getByTestId('Message');
    expect(renderedNotifications).toMatchSnapshot();
  });
  test('snapshot for info state matches', () => {
    render(<Message {...messages[1]} />);
    const renderedNotifications = screen.getByTestId('Message');
    expect(renderedNotifications).toMatchSnapshot();
  });
  test('snapshot for warn state matches', () => {
    render(<Message {...messages[2]} />);
    const renderedNotifications = screen.getByTestId('Message');
    expect(renderedNotifications).toMatchSnapshot();
  });
  test('snapshot for success state matches', () => {
    render(<Message {...messages[3]} />);
    const renderedNotifications = screen.getByTestId('Message');
    expect(renderedNotifications).toMatchSnapshot();
  });
  test('snapshot for standard state matches', () => {
    render(<Message {...messages[4]} />);
    const renderedNotifications = screen.getByTestId('Message');
    expect(renderedNotifications).toMatchSnapshot();
  });
});
