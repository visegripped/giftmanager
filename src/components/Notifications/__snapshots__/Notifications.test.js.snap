// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message snapshot for error state matches 1`] = `
<div
  class="message message--error"
  data-testid="Message"
>
  <div
    class="message--iconContainer"
  >
    <span
      class="message--icon"
    >
      ☠
    </span>
  </div>
  <pre
    class="message--pre "
  >
    this is an error notification.
  </pre>
  <div
    class="message--buttonContainer"
  >
    <button
      class="button primary button--error"
      data-testid="Button"
      id="1234-abcd-error"
    >
      <b>
        ✖
      </b>
    </button>
  </div>
</div>
`;

exports[`Message snapshot for info state matches 1`] = `
<div
  class="message message--info"
  data-testid="Message"
>
  <div
    class="message--iconContainer"
  >
    <span
      class="message--icon"
    >
      ⚙
    </span>
  </div>
  <pre
    class="message--pre "
  >
    this is an info notification.
  </pre>
  <div
    class="message--buttonContainer"
  >
    <button
      class="button primary button--info"
      data-testid="Button"
      id="1234-abcd-info"
    >
      <b>
        ✖
      </b>
    </button>
  </div>
</div>
`;

exports[`Message snapshot for standard state matches 1`] = `
<div
  class="message message--standard"
  data-testid="Message"
>
  <div
    class="message--iconContainer"
  >
    <span
      class="message--icon"
    >
      ☛
    </span>
  </div>
  <pre
    class="message--pre "
  >
    this is an standard notification.
  </pre>
  <div
    class="message--buttonContainer"
  >
    <button
      class="button primary button--standard"
      data-testid="Button"
      id="1234-abcd-standard"
    >
      <b>
        ✖
      </b>
    </button>
  </div>
</div>
`;

exports[`Message snapshot for success state matches 1`] = `
<div
  class="message message--success"
  data-testid="Message"
>
  <div
    class="message--iconContainer"
  >
    <span
      class="message--icon"
    >
      ☺
    </span>
  </div>
  <pre
    class="message--pre "
  >
    this is an success notification.
  </pre>
  <div
    class="message--buttonContainer"
  >
    <button
      class="button primary button--success"
      data-testid="Button"
      id="1234-abcd-success"
    >
      <b>
        ✖
      </b>
    </button>
  </div>
</div>
`;

exports[`Message snapshot for warn state matches 1`] = `
<div
  class="message message--warn"
  data-testid="Message"
>
  <div
    class="message--iconContainer"
  >
    <span
      class="message--icon"
    >
      ☹
    </span>
  </div>
  <pre
    class="message--pre "
  >
    this is an warn notification.
  </pre>
  <div
    class="message--buttonContainer"
  >
    <button
      class="button primary button--warn"
      data-testid="Button"
      id="1234-abcd-warn"
    >
      <b>
        ✖
      </b>
    </button>
  </div>
</div>
`;

exports[`Notifications snapshot matches 1`] = `
<section
  data-testid="Notifications"
/>
`;
